# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1349245946 14400
#      Wed Oct 03 02:32:26 2012 -0400
# Node ID da6c37ed0b4219a8d78ac3216c94ef1158db0640
# Parent  1865feebb41572e699c98debae28b8568842512f
eliminate calls to bind_internal_variable when setting --traditional options

* octave.cc (traditional): Delete static variable.
(maximum_braindamage): Call built-in functions directly instead of
using bind_internal_variable.
(octave_process_command_line): Handle traditional option directly.
(octave_initialize_interpreter): Don't handle traditional option here.

diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -41,16 +41,17 @@ along with Octave; see the file COPYING.
 #include "f77-fcn.h"
 #include "file-ops.h"
 #include "file-stat.h"
 #include "lo-error.h"
 #include "oct-env.h"
 #include "pathsearch.h"
 #include "str-vec.h"
 
+#include "builtins.h"
 #include "defaults.h"
 #include "Cell.h"
 #include "defun.h"
 #include "display.h"
 #include "error.h"
 #include "file-io.h"
 #include "help.h"
 #include "input.h"
@@ -108,20 +109,16 @@ static bool read_site_files = true;
 // TRUE means we set the initial path to configured defaults.
 // (--no-init-path)
 static bool set_initial_path = true;
 
 // TRUE means we don't print the usual startup message.
 // (--quiet; --silent; -q)
 static bool inhibit_startup_message = false;
 
-// TRUE means we turn on compatibility options.
-// (--traditional)
-static bool traditional = false;
-
 // If TRUE, print verbose info in some cases.
 // (--verbose; -V)
 static bool verbose_flag = false;
 
 // If TRUE, force the GUI to start.
 // (--force-gui)
 static bool force_gui_option = false;
 
@@ -662,30 +659,29 @@ initialize_error_handlers ()
 
 // What happens on --traditional.
 
 static void
 maximum_braindamage (void)
 {
   persist = true;
 
-  bind_internal_variable ("PS1", ">> ");
-  bind_internal_variable ("PS2", "");
-  bind_internal_variable ("PS4", "");
-  bind_internal_variable ("allow_noninteger_range_as_index", true);
-  bind_internal_variable ("beep_on_error", true);
-  bind_internal_variable ("confirm_recursive_rmdir", false);
-  bind_internal_variable ("crash_dumps_octave_core", false);
-  bind_internal_variable ("default_save_options", "-mat-binary");
-  bind_internal_variable ("do_braindead_shortcircuit_evaluation", true);
-  bind_internal_variable ("fixed_point_format", true);
-  bind_internal_variable ("history_timestamp_format_string",
-                         "%%-- %D %I:%M %p --%%");
-  bind_internal_variable ("page_screen_output", false);
-  bind_internal_variable ("print_empty_dimensions", false);
+  FPS1 (octave_value (">> "));
+  FPS2 (octave_value (""));
+  FPS4 (octave_value (""));
+  Fallow_noninteger_range_as_index (octave_value (true));
+  Fbeep_on_error (octave_value (true));
+  Fconfirm_recursive_rmdir (octave_value (false));
+  Fcrash_dumps_octave_core (octave_value (false));
+  Fdefault_save_options (octave_value ("-mat-binary"));
+  Fdo_braindead_shortcircuit_evaluation (octave_value (true));
+  Ffixed_point_format (octave_value (true));
+  Fhistory_timestamp_format_string (octave_value ("%%-- %D %I:%M %p --%%"));
+  Fpage_screen_output (octave_value (false));
+  Fprint_empty_dimensions (octave_value (false));
 
   disable_warning ("Octave:abbreviated-property-match");
   disable_warning ("Octave:fopen-file-in-path");
   disable_warning ("Octave:function-name-clash");
   disable_warning ("Octave:load-file-in-path");
   disable_warning ("Octave:possible-matlab-short-circuit-operator");
 }
 
@@ -857,17 +853,17 @@ octave_process_command_line (int argc, c
           break;
 
         case TEXI_MACROS_FILE_OPTION:
           if (optarg)
             texi_macros_file = optarg;
           break;
 
         case TRADITIONAL_OPTION:
-          traditional = true;
+          maximum_braindamage ();
           break;
 
         default:
           // getopt_long should print a message about unrecognized
           // options and return '?', which is handled above.  So if we
           // end up here, it is because there was an option but we
           // forgot to handle it.  That should be fatal.
           panic_impossible ();
@@ -991,19 +987,16 @@ octave_initialize_interpreter (int argc,
     command_editor::force_default_editor ();
 
   // These can come after command line args since none of them set any
   // defaults that might be changed by command line options.
 
   if (line_editing)
     initialize_command_input ();
 
-  if (traditional)
-    maximum_braindamage ();
-
   octave_interpreter_ready = true;
 
   initialize_version_info ();
 
   // Make all command-line arguments available to startup files,
   // including PKG_ADD files.
 
   intern_argv (argc, argv);
